### ISSUE TITLE
Incorrect Layout Resizing

### ISSUE THREAD

**Issue by JohnDoneth** [JohnDoneth/gleam-playground#4]
Zooming into the page or resizing the browser window does not re-layout the content as desired; requiring a full page refresh to appear proper.

<details>
  <summary>Examples</summary>

![image](https://user-images.githubusercontent.com/3239951/145912354-d851ec5e-af0b-4579-8485-ed21b6c68792.png)

![image](https://user-images.githubusercontent.com/3239951/145912420-b4bb23a3-7066-46ba-a4a0-425378c64ea4.png)
</details>

**Comment by inoas** [comment]
Yes, I noticed.
Related to this is that if you resize the window horizontally it does not rerender and content gets hidden.
On the other hand if you enlarge a window horizontally after load it behaves just fine.
I am not 100% certain but do think this was the case before my PR was merged.

**Comment by inoas** [comment]
Took a look but cannot figure out now why it did not work:
https://github.com/microsoft/monaco-editor/issues/28

tried this:

```
const gleamEditor = monaco.editor.create(document.getElementById("gleam-editor"), {
  value: source,
  language: "gleam",
  readOnly: false,
});

const jsEditor = monaco.editor.create(document.getElementById("javascript-editor"), {
  value: "// Click [Build & Run] to see JavaScript output here…",
  language: "javascript",
  readOnly: true,
});

const erlangEditor = monaco.editor.create(document.getElementById("erlang-editor"), {
  value: "// Click [Build] to see Erlang output here…",
  language: "erlang",
  readOnly: true,
});
```

```
window.addEventListener(
  "resize",
  function (event) {
    window.requestAnimationFrame(function () {
      console.log("resize layouts");
      gleamEditor.layout();
      jsEditor.layout();
      erlangEditor.layout();
    });
  },
  true
);
```

... however reflow of layout does nothing still for horizontal shrinking, only for horizontal growing it works.

**Comment by Aloso** [comment]
@JohnDoneth I fixed this in PR #10.
